src/prefect/cli/server.py

```diff
@@ -96,13 +101,63 @@ def server():
 @click.option(
     "--no-ui", "-u", help="Pass this flag to avoid starting the UI", is_flag=True,
 )
-def start(version, skip_pull, no_upgrade, no_ui):
+@click.option(
+    "--postgres-port",
+    help="The port used to serve Postgres",
+    default=config.server.database.host_port,
+    type=str,
+)
+@click.option(
+    "--hasura-port",
+    help="The port used to serve Hasura",
+    default=config.server.hasura.host_port,
+    type=str,
+)
+@click.option(
+    "--graphql-port",
+    help="The port used to serve the GraphQL API",
+    default=config.server.graphql.host_port,
+    type=str,
+)
+@click.option(
+    "--ui-port",
+    help="The port used to serve the UI",
+    default=config.server.ui.port,
+    type=str,
+)
+@click.option(
+    "--server-port",
+    help="The port used to serve the Core server",
+    default=config.server.port,
+    type=str,
+)
+def start(
+    version,
+    skip_pull,
+    no_upgrade,
+    no_ui,
+    postgres_port,
+    hasura_port,
+    graphql_port,
+    ui_port,
+    server_port,
+):
     """
     This command spins up all infrastructure and services for Prefect Server
     """
     docker_dir = Path(__file__).parents[0]
 
-    env = make_env()
+    # Temporary config set for port allocation
+    with set_temporary_config(
+        {
+            "server.database.host_port": postgres_port,
+            "server.hasura.host_port": hasura_port,
+            "server.graphql.host_port": graphql_port,
+            "server.ui.port": ui_port,
+            "server.port": server_port,
```

Comment:
Any reason not to make them all `host_port`?

Reply:
I can make them all host port but then it might get confusing for the Apollo API endpoint. Because if you run it exposing the endpoint at a different port then `server.port` then you will also need to update your `server.port` in order for prefect to communicate with it

Reply:
But I see the point in making it host port for all because that the purpose it serves here